src/commands/init/index.ts

```diff
@@ -47,26 +58,78 @@ export default class Init extends Command {
             required: false,
             default: false,
         }),
+        package: Flags.string({
+            char: 'p',
+            aliases: ['package'],
+            description: '',
+            required: false,
+            default: '',
+        }),
     };
 
     async run(): Promise<void> {
         const { flags } = await this.parse(Init);
         this.log(`Initializing Velocitas packages ...`);
-        const projectConfig = this.initializeOrReadProject();
+        const projectConfig = this._initializeOrReadProject();
 
         const appManifestData = AppManifest.read(projectConfig.getVariableMappings().get(APP_MANIFEST_PATH_VARIABLE));
 
-        await this.ensurePackagesAreDownloaded(projectConfig, flags.force, flags.verbose);
-        projectConfig.validateUsedComponents();
+        let packageConfigs = projectConfig.getPackages(false);
+        if (flags.package !== '') {
+            const requestedPackageConfig = this._parsePackageConfig(flags.package);
+            await this._resolveVersion(requestedPackageConfig, flags.verbose);
+            await this._modifyProjectConfig(requestedPackageConfig, projectConfig, flags.force, flags.verbose);
+            await this._ensurePackageIsDownloaded(requestedPackageConfig, flags.force, flags.verbose);
+            this._addComponentsToProjectConfig(requestedPackageConfig, projectConfig);
+            // projectConfig.validateUsedComponents();
+        } else {
+            for (const packageConfig of packageConfigs) {
+                await this._resolveVersion(packageConfig, flags.verbose);
+                await this._ensurePackageIsDownloaded(packageConfig, flags.force, flags.verbose);
+            }
+            projectConfig.validateUsedComponents();
+        }
 
         if (!flags['no-hooks']) {
-            await this.runPostInitHooks(projectConfig, appManifestData, flags.verbose);
+            await this._runPostInitHooks(projectConfig, appManifestData, flags.verbose);
         }
 
-        this.createProjectLockFile(projectConfig, flags.verbose);
+        projectConfig.write();
+        this._createProjectLockFile(projectConfig, flags.verbose);
     }
 
-    initializeOrReadProject(): ProjectConfig {
+    private _addComponentsToProjectConfig(requestedPackageConfig: PackageConfig, projectConfig: ProjectConfig) {
+        if (requestedPackageConfig !== null) {
+            const providedComponents = requestedPackageConfig.readPackageManifest().components;
+            const areComponentsExisting = providedComponents.some((comp) => {
+                const enabledComponents = projectConfig.getComponentsFromInstalledPackages().map((comp) => comp.config.id);
+                return enabledComponents.indexOf(comp.id) > -1;
+            });
+
+            if (!areComponentsExisting) {
+                providedComponents.forEach((providedComponent) => {
+                    projectConfig.addComponent(providedComponent.id);
+                });
+            }
```

Comment:
Adding all components if no component of a package is found. However, this approach does not cover when a single component is added / removed during an update.